server/src/repositories/board.repository.ts

```diff
@@ -1,19 +1,34 @@
 import { EntityRepository, getCustomRepository, Repository } from 'typeorm';
 import { Board } from '../entity/Board';
-import {UserRepository} from './user.repository';
+import { UserRepository } from './user.repository';
+import { IBoardModel, IReducedBoard } from '../models/Board';
 
 @EntityRepository(Board)
 export class BoardRepository extends Repository<Board> {
 	findByType(boardType: string) {
 		return this.findOneOrFail({ where: { boardType } });
 	}
 
-	getAll() {
-		return this.find();
-	}
+	getAll = async (): Promise<IBoardModel[]> => {
+		const extendedBoards = await this.find().then(boards =>
```

Comment:
Should be handled using one query. Can be like getOne

Reply:
Can this query be changed to something like this?
`	.createQueryBuilder('board')
			.innerJoin('board.createdBy', 'user')
			.addSelect('user.id')
			.addSelect('user.firstName')
			.addSelect('user.lastName')
			.addSelect('user.avatar')
			.getMany();`